src/averageAges.js

```diff
@@ -55,7 +63,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    onlyWithSon
+      ? people.find(mam =>
+        mam.name === person.mother) && person.sex === 'm'
+      : people.find(mam => mam.name === person.mother));
```

Comment:
Let's create a variable for the `people.find(mam => mam.name === person.mother)` expression